x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/results/SparseEmbeddingResults.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.results;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.inference.InferenceResults;
+import org.elasticsearch.inference.InferenceServiceResults;
+import org.elasticsearch.inference.TaskType;
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.core.ml.inference.results.TextExpansionResults;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.elasticsearch.xpack.core.ml.inference.trainedmodel.InferenceConfig.DEFAULT_RESULTS_FIELD;
+
+public record SparseEmbeddingResults(List<Embedding> embeddings, boolean isTruncated) implements InferenceServiceResults {
+
+    public static final String NAME = "sparse_embedding_results";
+    public static final String SPARSE_EMBEDDING = TaskType.SPARSE_EMBEDDING.toString();
+    public static final String EMBEDDING = "embedding";
+    public static final String IS_TRUNCATED = "is_truncated";
+
+    public SparseEmbeddingResults(StreamInput in) throws IOException {
+        this(in.readCollectionAsList(Embedding::new), in.readBoolean());
+    }
+
+    public static SparseEmbeddingResults create(List<? extends InferenceResults> results) {
+        boolean isTruncated = false;
+        List<Embedding> embeddings = new ArrayList<>(results.size());
+
+        for (InferenceResults result : results) {
+            if (result instanceof TextExpansionResults expansionResults) {
+                isTruncated |= expansionResults.isTruncated();
```

Comment:
If for some reason only 1 of the results is truncated we'll mark them all as truncated.